server/src/tests/Trade.test.ts

```diff
@@ -0,0 +1,78 @@
+import { Trade } from "../Trade";
+import SocketMock from "socket.io-mock-ts";
+import { textSpanIntersectsWithTextSpan } from "typescript";
+import { TetrominoType } from "../../../common/TetrominoType";
+
+describe("Testing 'Trade'", () => {
+    let clientSocket1: any;
+    let clientSocket2: any;
+    let randTrader: Trade;
+    let trader: Trade;
+
+    beforeAll(() => {
+        clientSocket1 = new SocketMock();
+        clientSocket2 = new SocketMock();
+        // clientSocket1.on("sendRandomPiece", (tetromino: TetrominoType) => {
+        //     client1Tetromino = tetromino;
+        // });
+        // clientSocket1.on("sendTradePiece", (tetromino: TetrominoType) => {
+        //     client2Tetromino = tetromino;
+        // });
+        // clientSocket2.on("sendRandomPiece", () => {});
+        // clientSocket2.on("sendTradePiece", () => {});
+    });
+    beforeEach(() => {
+        randTrader = new Trade(1);
+        trader = new Trade();
+    });
+    test("First Offer for regular trader", () => {
```

Comment:
For all these test cases, if they fulfill *some* FR on the SRS, can you include it into the test name (like how we did it for the other server tests & client tests)
For example, this test case partially tests FR21